src/app/facebook-pixel-service/facebook-pixel-provider.ts

```diff
@@ -0,0 +1,29 @@
+import { Provider, APP_BOOTSTRAP_LISTENER, ComponentRef } from '@angular/core';
+import { Router, NavigationEnd } from '@angular/router';
+import { FacebookPixelService } from './facebook-pixel-service';
+
+export const FacebookPixelProvider: Provider = {
```

Comment:
@tbg-jasonfung this seems like an odd and over-complicated way to init the service.

We also haven't accounted for the fact that server-side there is no `window`.

Have a look at where we call these in `AppComponent` – we need to do something analogous for the FB code.

```
      this.analyticsService.init();
      this.getSiteControlService.init();
```